src/Orleans/Logging/ILoggerExtensions.cs

```diff
@@ -0,0 +1,274 @@
+        public static void Trace(this ILogger logger, string format, params object[] args)
+        {
+            logger.LogTrace(format, args);
+        }
+
+        /// <summary>
+        /// Writes a log entry at the Verbose2 severity level.
+        /// Verbose2 is lower than Verbose.
+        /// </summary>
+        /// <param name="logger">The logger</param>
+        /// <param name="message">The log message.</param>
+        public static void Trace(this ILogger logger, string message)
+        {
+            logger.LogTrace(message);
+        }
+
+        /// <summary>
+        /// Writes a log entry at the Information Level
+        /// </summary>
+        /// <param name="logger">Target logger.</param>
+        /// <param name="format">A standard format string, suitable for String.Format.</param>
+        /// <param name="args">Any arguments to the format string.</param>
+        public static void Info(this ILogger logger, string format, params object[] args)
+        {
+            logger.LogInformation(format, args);
+        }
+
+        /// <summary>
+        /// Writes a log entry at the Info logLevel 
+        /// </summary>
+        /// <param name="logger">Target logger.</param>
+        /// <param name="message">The log message.</param>
+        public static void Info(this ILogger logger, string message)
+        {
+            logger.LogInformation(message);
+        }
+
+        /// <summary>
+        /// Writes a log entry at the Debug logLevel
+        /// </summary>
+        /// <param name="logger">The logger</param>
+        /// <param name="logCode">The log code associated with this message.</param>
+        /// <param name="format">A standard format string, suitable for String.Format.</param>
+        /// <param name="args">Any arguments to the format string.</param>
+        public static void Debug(this ILogger logger, int logCode, string format, params object[] args)
+        {
+            logger.LogDebug(LoggingUtils.CreateEventId(0, logCode), format, args);
+        }
+
+        internal static void Debug(this ILogger logger, ErrorCode logCode, string format, params object[] args)
+        {
+            logger.LogDebug(LoggingUtils.CreateEventId(0, logCode), format, args);
+        }
+
+        /// <summary>
+        /// Writes a log entry at the Debug logLevel
+        /// </summary>
+        /// <param name="logger">The logger</param>
+        /// <param name="logCode">The log code associated with this message.</param>
+        /// <param name="message">The log message.</param>
+        public static void Debug(this ILogger logger, int logCode, string message)
+        {
+            logger.LogDebug(LoggingUtils.CreateEventId(0, logCode), message);
+        }
+
+        internal static void Debug(this ILogger logger, ErrorCode logCode, string message)
+        {
+            logger.LogDebug(LoggingUtils.CreateEventId(0, logCode), message);
```

Comment:
why are you passing `0` in all of these calls? That was the important piece of information, not just the string for the error code.
Shouldn't the ID be inferred from the ErrorCode instead?

Reply:
I was under the impression that event id can be event sequence id or errorcode, which indicating an event type. So I left event id as 0, just in case we want to use it as sequence id in the future. But I maybe under a wrong impression. will change it.